stdlib/src/mmap.rs

```diff
@@ -0,0 +1,1036 @@
+            let file_len = match file.metadata() {
+                Ok(m) => m.len(),
+                Err(e) => return Err(vm.new_os_error(e.to_string())),
+            };
+
+            Ok(PyInt::from(file_len).into_ref(vm))
+        }
+
+        #[pymethod]
+        fn tell(&self, vm: &VirtualMachine) -> PyResult<PyIntRef> {
+            Ok(PyInt::from(self.inner_pos()).into_ref(vm))
+        }
+
+        #[pymethod]
+        fn write(&self, bytes: ArgBytesLike, vm: &VirtualMachine) -> PyResult<PyIntRef> {
+            let pos = self.inner_pos();
+            let size = self.inner_size();
+
+            let data = bytes.borrow_buf();
+
+            if pos > size || size - pos < data.len() as isize {
+                return Err(vm.new_value_error("data out of range".to_owned()));
+            }
+
+            let len = self.try_writable(vm, |mmap| {
+                (&mut mmap[pos as usize..(pos as usize + data.len())])
+                    .write(&data)
+                    .map_err(|e| vm.new_os_error(e.to_string()))?;
+                Ok(data.len())
+            })??;
+
+            self.advance_pos(len as isize);
+
+            Ok(PyInt::from(len).into_ref(vm))
+        }
+
+        #[pymethod]
+        fn write_byte(&self, byte: PyObjectRef, vm: &VirtualMachine) -> PyResult<()> {
+            let b = value_from_object(vm, &byte)?;
+
+            let pos = self.inner_pos();
+            let size = self.inner_size();
+
+            if pos >= size {
+                return Err(vm.new_value_error("write byte out of range".to_owned()));
+            }
+
+            self.try_writable(vm, |mmap| {
+                mmap[pos as usize] = b;
+            })?;
+
+            self.advance_pos(1);
+
+            Ok(())
+        }
+
+        fn get_item_by_index(&self, i: isize, vm: &VirtualMachine) -> PyResult<PyObjectRef> {
+            let i = wrap_index(i, self.len())
+                .ok_or_else(|| vm.new_index_error("mmap index out of range".to_owned()))?;
+
+            let b = match self.check_valid(vm)?.deref().as_ref().unwrap() {
+                MmapObj::Read(mmap) => mmap[i as usize],
+                MmapObj::Write(mmap) => mmap[i as usize],
+            };
+
+            Ok(PyInt::from(b).into_ref(vm).into())
+        }
+
+        fn getitem_by_slice(
+            &self,
+            slice: &SaturatedSlice,
+            vm: &VirtualMachine,
+        ) -> PyResult<PyObjectRef> {
+            let (range, step, slicelen) = slice.adjust_indices(self.len());
+
+            let mmap = self.check_valid(vm)?;
+
+            if slicelen == 0 {
+                return Ok(PyBytes::from(vec![]).into_ref(vm).into());
+            } else if step == 1 {
+                let bytes = match mmap.deref().as_ref().unwrap() {
+                    MmapObj::Read(mmap) => &mmap[range],
+                    MmapObj::Write(mmap) => &mmap[range],
+                };
+                return Ok(PyBytes::from(bytes.to_vec()).into_ref(vm).into());
```

Comment:
```suggestion
                return Ok(PyBytes::from(bytes.to_vec()));
```

Reply:
> 

Please review the lastest commit https://github.com/RustPython/RustPython/pull/3755/commits/46453dcd8e877282582446541a50636c37be8034

If you think the PR is ready to merge, i will do the rebase job.

Reply:
> Yes, please rebase it if you can do.

Please review the lastest commit https://github.com/RustPython/RustPython/pull/3755/commits/46453dcd8e877282582446541a50636c37be8034

If you think the PR is ready to merge, i will do the rebase job.